src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java

```diff
@@ -0,0 +1,159 @@
+                break;
+            case PAST:
+                bookings = bookingRepository.findAllByBooker_IdAndEndIsBefore(userId, LocalDateTime.now(), sort);
+                break;
+            case FUTURE:
+                bookings = bookingRepository.findAllByBooker_IdAndStartIsAfter(userId, LocalDateTime.now(), sort);
+                break;
+            case CURRENT:
+                bookings = bookingRepository
+                        .findAllByBooker_IdAndStartIsBeforeAndEndIsAfter(
+                                userId, LocalDateTime.now(), LocalDateTime.now(), sort);
+                break;
+            case WAITING:
+                bookings = bookingRepository.findAllByBooker_IdAndStatus(userId, Status.WAITING);
+                break;
+            case REJECTED:
+                bookings = bookingRepository.findAllByBooker_IdAndStatus(userId, Status.REJECTED);
+                break;
+        }
+
+        return bookings.isEmpty() ? Collections.emptyList() : bookings.stream()
+                .map(BookingMapper::toBookingInfoDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<BookingInfoDto> getByOwner(Long userId, String value) throws UserNotFoundException, InvalidStatusException {
+        State state = validateState(value);
+        User owner = userRepository.findById(userId).orElseThrow(() -> new UserNotFoundException("user not found"));
+        List<Booking> bookings = new ArrayList<>();
+        Sort sort = Sort.by(Sort.Direction.DESC, "start");
+        switch (state) {
+            case ALL:
+                bookings = bookingRepository.findAllByItem_Owner_IdOrderByStartDesc(userId);
+                break;
+            case PAST:
+                bookings = bookingRepository.findAllByItem_Owner_IdAndEndIsBefore(userId, LocalDateTime.now(), sort);
+                break;
+            case FUTURE:
+                bookings = bookingRepository.findAllByItem_Owner_IdAndStartIsAfter(userId, LocalDateTime.now(), sort);
+                break;
+            case CURRENT:
+                bookings = bookingRepository
+                        .findAllByItem_Owner_IdAndStartIsBeforeAndEndIsAfter(
+                                userId, LocalDateTime.now(), LocalDateTime.now(), sort);
+                break;
+            case WAITING:
+                bookings = bookingRepository.findAllByItem_Owner_IdAndStatus(userId, Status.WAITING);
+                break;
+            case REJECTED:
+                bookings = bookingRepository.findAllByItem_Owner_IdAndStatus(userId, Status.REJECTED);
+                break;
+        }
+
+        return bookings.isEmpty() ? Collections.emptyList() : bookings.stream()
+                .map(BookingMapper::toBookingInfoDto)
+                .collect(Collectors.toList());
+    }
+
+    private State validateState(String value) throws InvalidStatusException {
+        State state = State.ALL;
+        try {
+            state = State.valueOf(value);
```

Comment:
Лучше сразу возвращать результат: 
```suggestion
            return State.valueOf(value);
```
Тогда не нужно будет вводить переменную `state`